java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -0,0 +1,2943 @@
+/*
+ *
+ *  Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static ai.rapids.cudf.HostColumnVector.OFFSET_SIZE;
+
+/** This class represents the column_view of a column analogous to its cudf cpp counterpart.
+ * It holds view information like the native handle and other metadata for a column_view. It also
+ * exposes APIs that would allow operations on a view.
+ *
+ */
+public class ColumnView implements AutoCloseable, BinaryOperable {
+
+  public static final long UNKNOWN_NULL_COUNT = -1;
+
+  protected long viewHandle = 0;
+  protected DType type;
+  protected long rows;
+  protected Optional<Long> nullCount = Optional.empty();
+
+
+  //TODO can we make this better
+  protected ColumnView(long address) {
+    this.viewHandle = address;
+    this.type = DType.fromNative(ColumnView.getNativeTypeId(getNativeView()), ColumnView.getNativeTypeScale(getNativeView()));
```

Comment:
nit: we just set the viewHandle, so why do we call a getter 4 times to get it again? If the getter is important to use then lets cache it, or just go off of address here in the constructor.